### PR TITLE
implementing read_logs for postgres

### PR DESCRIPTION
**Pull request by arthurmm-cloudwalk**
Implementing get logs for postgres

### CODE REVIEW

File: src/eth/storage/postgres/postgres.rs [hunk 1]
```diff
@@ -308,8 +311,55 @@ impl EthStorage for Postgres {
...
+        if let Some(block_number) = filter.to_block {
+            builder.push(&" AND block_number <= $2");
+            to = block_number.try_into().unwrap();
+            builder.push_bind(to.to_owned());
+        }
+        
+        let query = builder.build();
+        let result = query
+            .map(|row: PgRow| LogMined {
+                log: crate::eth::primitives::Log {
```
Comment (arthur-cw) [inline src/eth/storage/postgres/postgres.rs:344]: Move the import to the file header

File: src/eth/storage/postgres/postgres.rs [hunk 2]
```diff
@@ -308,8 +311,55 @@ impl EthStorage for Postgres {
...
+                log_index: row.get("log_idx"),
+                block_number: row.get("block_number"),
+                block_hash: row.get("block_hash"),
+            })
+	    	.fetch_all(&self.connection_pool)
+		    .await?
+            .iter()
+            .filter(|log| filter.matches(log))
+            .cloned()
+            .collect();
```
Comment (arthur-cw) [inline src/eth/storage/postgres/postgres.rs:360]: change to `into_iter()` so you don't have to clone after the filter.
```suggestion
            .into_iter()
            .filter(|log| filter.matches(log))
            .collect();
```

File: src/eth/storage/postgres/postgres.rs [hunk 3]
```diff
@@ -308,8 +311,53 @@ impl EthStorage for Postgres {
...
+            WHERE block_number >= $1
+        "#
+        .to_owned();
+        let builder = &mut QueryBuilder::new(query);
+        builder.push_bind(from);
+
+        // verifies if to_block exists
+        if let Some(block_number) = filter.to_block {
+            builder.push(" AND block_number <= $2");
+            to = block_number.try_into().unwrap();
```
Comment (arthur-cw) [inline src/eth/storage/postgres/postgres.rs:336]: use `?` instead of `unwrap()`
```suggestion
            to = block_number.try_into()?;
```

Comment (arthurmm-cloudwalk) [reply src/eth/storage/postgres/postgres.rs]: Done

File: src/eth/storage/postgres/postgres.rs (hunk 2 above)
Comment (arthurmm-cloudwalk) [reply src/eth/storage/postgres/postgres.rs]: Done

File: src/eth/storage/postgres/postgres.rs (hunk 1 above)
Comment (arthurmm-cloudwalk) [reply src/eth/storage/postgres/postgres.rs]: Done

File: src/eth/storage/postgres/postgres.rs [hunk 4]
```diff
@@ -308,8 +312,51 @@ impl EthStorage for Postgres {
...
+                address as "address:  "
+                , data as "data:  "
+                , transaction_hash as "transaction_hash: "
+                , log_idx as "log_idx:  "
+                , block_number as "block_number:  "
+                , block_hash as "block_hash:  "
+            FROM logs
+            WHERE block_number >= $1
+        "#
+        .to_owned();
```
Comment (arthur-cw) [inline src/eth/storage/postgres/postgres.rs:328]: I think this is not necessary
```suggestion
```

File: src/eth/storage/postgres/postgres.rs [hunk 5]
```diff
@@ -308,8 +312,51 @@ impl EthStorage for Postgres {
...
+        let builder = &mut QueryBuilder::new(query);
+        builder.push_bind(from);
+
+        // verifies if to_block exists
+        if let Some(block_number) = filter.to_block {
+            builder.push(" AND block_number <= $2");
+            let to: i64 = block_number.try_into()?;
+            builder.push_bind(to.to_owned());
```
Comment (arthur-cw) [inline src/eth/storage/postgres/postgres.rs:336]: also unecessary
```suggestion
            builder.push_bind(to);
```

Comment (arthurmm-cloudwalk) [reply src/eth/storage/postgres/postgres.rs]: Done

File: src/eth/storage/postgres/postgres.rs (hunk 4 above)
Comment (arthurmm-cloudwalk) [reply src/eth/storage/postgres/postgres.rs]: Done

**Review by arthur-cw — APPROVED** [review]

**Pull request merged**